### ISSUE TITLE
Error reading a Sigma ride .fit file

### ISSUE THREAD

**Issue by sje30** [grimbough/FITfileR#40]
Hi Mike,

Thanks for the package.

I hit an error with trying to use this package importing a file from my Sigma Rox 11.1 bike computer.  It seems like the file is fine (it renders okay on https://www.fitfileviewer.com ).  Not expecting you to fix this (I can work around it seems by going online to sigma and exporting it in a different format, such as .slf which looks like xml) but if you think its fixable from the backtrace below, I can start looking.

(I could share the file with you via email as it probably has personal info in it; or I can make a short test file for sharing).

Stephen

```
> dat <- readFitFile("2024-04-20_20-53-47_Road Bike.fit")
Error in names(fields) <- c("field_def_num", "size", "base_type") : 
  'names' attribute [3] must be the same length as the vector [2]
> traceback()
4: .processFieldDefs(readBin(con = con, what = "raw", n = 3 * n_fields, 
       size = 1, signed = FALSE))
3: .readMessage_definition(con = con, message_header = record_header)
2: .readFile(fileName, preallocate = TRUE)
1: readFitFile("2024-04-20_20-53-47_Road Bike.fit")
```

```
> sessionInfo()
R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin23.2.0 (64-bit)
Running under: macOS Sonoma 14.4.1

Matrix products: default
BLAS:   /opt/homebrew/Cellar/openblas/0.3.26/lib/libopenblasp-r0.3.26.dylib 
LAPACK: /opt/homebrew/Cellar/r/4.3.3/lib/R/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] FITfileR_0.1.8

loaded via a namespace (and not attached):
 [1] desc_1.4.3       utf8_1.2.4       R6_2.5.1         bit_4.0.5        tidyselect_1.2.1 magrittr_2.0.3  
 [7] remotes_2.5.0    glue_1.7.0       tibble_3.2.1     pkgconfig_2.0.3  bit64_4.0.5      dplyr_1.1.4     
[13] generics_0.1.3   lifecycle_1.0.4  ps_1.7.6         cli_3.6.2        fansi_1.0.6      processx_3.8.4  
[19] callr_3.7.5      vctrs_0.6.5      compiler_4.3.3   tools_4.3.3      pkgbuild_1.4.4   curl_5.2.1      
[25] pillar_1.9.0     rlang_1.1.3     
> 
```

**Comment by sje30** [comment]
A shorter file I can share here (zipped so that GitHub accepts it) that has the same error:

```
> library(FITfileR)
> dat <- readFitFile("~/2024_04_22__10_20_RoadBike.fit")
Error in names(fields) <- c("field_def_num", "size", "base_type") : 
  'names' attribute [3] must be the same length as the vector [2]

```

and yet loads in fitfileviewer.com

<img width="1264" alt="Screenshot 2024-04-22 at 17 59 25" src="https://github.com/grimbough/FITfileR/assets/1095067/b2d4b2b0-2ac3-4a56-ad92-4b1c444dfbc0">

[example.zip](https://github.com/grimbough/FITfileR/files/15066308/example.zip)

**Comment by grimbough** [comment]
Thanks for the report and sharing the file.  It looks like this device defines some messages that have no content, so FITfileR was trying to process a vector of length 0.  I've patched this in 0.1.9 and it looks like it works ok now:

```r
library(FITfileR)

## download and extract the example file
destfile = tempfile(fileext = ".zip")
url = "https://github.com/grimbough/FITfileR/files/15066308/example.zip"
dl = download.file(url = url, destfile = destfile)
example_file = unzip(destfile, exdir = tempdir())

## read and extract the file ID
fitFile <- readFitFile(example_file[1])
FITfileR::file_id(fitFile)
#> # A tibble: 1 × 5
#>   type     manufacturer product time_created        product_name
#>   <chr>    <chr>          <int> <dttm>              <chr>       
#> 1 activity sigmasport        49 2024-04-22 09:20:20 ROX 11.1 EVO
```
